### DISCUSSION TITLE
Proper Chaining and sequence of plugins

### DISCUSSION THREAD

**Discussion by Marcel2508** [coredns/coredns#5252] [Q&A] [answered]
Hi, I'm trying to write a custom plugin based on the https://github.com/coredns/demo plugin.
I want to return different results for different source IPs.
But the Problem is coredns seems to ignore my custom plugin when I'm using it together with forward plugin:

```
.:53 {
    log
    serversdns    <--- my custom Plugin
    forward . 1.1.1.1
}
```
My plugin is loaded on start but ``ServeDNS`` never get called.

When using this config without forward plugin, my plugin runs as expected and ``ServeDNS`` gets called as expected.
But I need the DNS Server to resolve domains other than my custom ones aswell..

Any Idea what could cause this problem and how to resolve this? 

Edit: I cannot put my plugin inside a domain block, since some subdomains require custom responses and some are public (need forwarding)

Thanks in advance

**Comment by Tantalor93** [comment, marked as answer]
Hello, in order to load your plugin to your custom CoreDNS and set the correct order you need to [override the order of plugins using directives](https://github.com/coredns/coredns/blob/master/core/dnsserver/zdirectives.go#L12), for example like this

```
var directives = []string{
	"metadata",
	"debug",
	"trace",
	"ready",
	"health",
	"pprof",
	"prometheus",
	"log",
	"any",
	"serversdns" // your plugin
	"forward"
}

func init() {
  dnsserver.Directives = directives
}

func main() {
	coremain.Run()
}
```

**Reply by Marcel2508** [reply]
Ahh great thanks for the fast reply! I didn't knew this. Now it works as expected

**Comment by chrisohaver** [comment]
The more automated way to do this is via `plugin.cfg`
